### PR TITLE
Mutable external ID

### PR DESCRIPTION
**Pull request by alxdarksage**
Also removing the "allowIdentifierUpdates" in the account DAO because we always copy user input over from a StudyParticipant to an account from the database, so there's no danger of these being changed other than where we change them in our own code (by intention).

NOTE: I will put the exception handling fixes and improvements from the other PR into a separate PR.

### CODE REVIEW

File: app/org/sagebionetworks/bridge/services/ConsentPdf.java [hunk 1]
```diff
@@ -106,11 +106,17 @@ private String createSignedDocument() {
...
+            // exposed this bug.
             String userEmail = signer.getEmail();
             String html = studyConsentAgreement.replace("@@name@@", username);
             html = html.replace("@@signing.date@@", signingDate);
-            html = html.replace("@@email@@", userEmail);
+            if (signer.getEmail() != null && Boolean.TRUE.equals(signer.getEmailVerified())) {
+                html = html.replace("@@email@@", userEmail);
+            } else {
+                html = html.replace("@@email@@", "");
+            }
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/services/ConsentPdf.java:119]: Was this intended to be part of this pull request? If so, should we add unit tests for this?

File: test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java [hunk 2]
```diff
@@ -30,12 +30,12 @@
 import org.joda.time.DateTime;
 import org.joda.time.DateTimeZone;
 import org.junit.Before;
+//import org.junit.Ignore;
```
Comment (DwayneJengSage) [inline test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java]: Nitpick: Remove commented import.

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]

File: app/org/sagebionetworks/bridge/services/ConsentPdf.java (hunk 1 above)
Comment (alxdarksage) [reply app/org/sagebionetworks/bridge/services/ConsentPdf.java:119]: Yes, integration tests run against my consents as fixed locally, ran a text against a legacy document when there was no email address, and this generates an NPE. (I don't think any of these legacy documents are in production.) Added a test for this.

File: test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java (hunk 2 above)
Comment (alxdarksage) [reply test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java]: Removed

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**